$OpenBSD: patch-capplets_common_gtkrc-utils_c,v 1.1 2009/04/13 14:35:19 ajacoutot Exp $

Default number of open file descriptors is too low on OpenBSD.

--- capplets/common/gtkrc-utils.c.orig	Wed Sep 24 18:36:22 2008
+++ capplets/common/gtkrc-utils.c	Mon Apr 13 16:13:01 2009
@@ -33,6 +33,28 @@
 #define ENGINE_SYMBOL ((gpointer) 2)
 #define COLOR_SCHEME_SYMBOL ((gpointer) 3)
 
+#ifdef __OpenBSD__
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+
+static int
+fdlim_set(int lim)
+{ 
+	struct rlimit rlfd;
+ 
+	if (lim <= 0 || lim > FD_SETSIZE)
+	    return (-1);
+	if (getrlimit(RLIMIT_NOFILE, &rlfd) < 0)
+	    return (-1);
+	if (rlfd.rlim_cur < lim)
+	    rlfd.rlim_cur = lim;
+	if (setrlimit(RLIMIT_NOFILE, &rlfd) < 0)
+	    return (-1);
+	return (0);
+}
+#endif // __OpenBSD__
+
 gchar *
 gtkrc_find_named (const gchar *name)
 {
@@ -104,6 +126,9 @@ gtkrc_get_details (gchar *filename, GSList **engines, 
 
 		read_files = g_slist_prepend (read_files, filename);
 
+#ifdef __OpenBSD__
+		fdlim_set(256);
+#endif
 		file = g_open (filename, O_RDONLY);
 		if (file == -1)
 		{
@@ -198,6 +223,9 @@ gtkrc_get_color_scheme (const gchar *gtkrc_file)
 
 		read_files = g_slist_prepend (read_files, filename);
 
+#ifdef __OpenBSD__
+		fdlim_set(256);
+#endif
 		file = g_open (filename, O_RDONLY);
 		if (file == -1)
 		{
